.changeset/metal-towns-flow.md

```diff
@@ -0,0 +1,50 @@
+---
+'@shopify/shopify-app-react-router': minor
+---
+
+We are introducing support for expiring offline access tokens. This feature improves security by limiting the lifespan of offline access tokens and automatically refreshing them using refresh tokens.
+
+- **New Future Flag**: Added `expiringOfflineAccessTokens` (boolean) to the `future` configuration in `shopifyApp`. When enabled, the library will start using expiring offline tokens and automatically check if it is expired or nearing expiration. If expired/expiring, it attempts to refresh the access token using the stored refresh token. Defaults to `false` for backward compatibility.
+
+- **Automatic Token Refresh**: Integrated token refresh logic into authentication flows (`flow`, `fulfillmentService`, `appProxy`, `webhooks`) and unauthenticated contexts (`admin`, `storefront`). When a session is loaded and found to be expired (or expiring within 1 second), and the feature is enabled, the library transparently refreshes the token and persists the new session data.
+
+To enable expiring offline access tokens in your app, you must ensure your session storage can persist refresh tokens. If you are using Prisma, follow these steps:
```

Comment:
I think we need to make it really obvious and really explicit that this will only work for Prisma.  E.g:

For now this will only work if you are using the Prisma Session Storage package. We will be updating the other session storage packages in the coming [TIMEFRAME]